components/Terminal/index.vue

```diff
@@ -1,150 +1,193 @@
 <script setup lang="ts">
-import { ref } from 'vue'
+import { ref } from "vue";
 
-import TerminalBlinker from './TerminalBlinker.vue'
-import TerminalHistoryItem from './TerminalHistoryItem.vue'
-import TerminalMarker from './TerminalMarker.vue'
-
-import get_command from './commands'
-import { cwd } from './commands/filesystem'
+import TerminalBlinker from "./TerminalBlinker.vue";
+import TerminalHistoryItem from "./TerminalHistoryItem.vue";
+import TerminalMarker from "./TerminalMarker.vue";
+import type { State } from "./commands/registry";
 
+import get_command from "./commands";
+import { cwd } from "./commands/filesystem";
+import register from "./commands/registry";
 
 interface HistoryItem {
-    input: string,
-    output: string | undefined,
-    cwd: string,
+  input: string;
+  output: string | undefined;
+  cwd: string;
 }
 
+const inputBuffer = ref("");
+const history = ref<HistoryItem[]>([]);
+const activeLineBuffer = ref("");
+const historySelectionOffset = ref(0);
 
-const inputBuffer = ref('')
-const history = ref<HistoryItem[]>([])
-const activeLineBuffer = ref('')
-const historySelectionOffset = ref(0)
-
-const commandState = ref<{ [key: string]: any }>({})
-cwd(commandState.value, [])
-
+// TODO: We could think about persisting the state in the future
+// I would prefer if we didnt hardcode the initial value and instead called `userHome`,
+// but the issue is that it depends on this variable, hence introducing a circular dependency
+const commandState = ref<State>({ filesystem: { cwd: "/home/anonymous" } });
```

Comment:
This was the reason for the `cwd` on the following line. Non-logged-in users should be anonymous (and all users start out logged in at some point) so I don't think calling `userHome` would be necessary, especially if we later persist the state for logged in users.